mmv1/third_party/terraform/fwresource/with_get_project.go

```diff
@@ -0,0 +1,39 @@
+package fwresource
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+type ProjectGetter interface {
+	GetProject(types.String, *transport_tpg.Config, *diag.Diagnostics) types.String
+}
+
+type WithGetProject struct {
+}
+
+// GetProject determines whether to use a project value from the resource/data source config or the provider-default project
+func (w *WithGetProject) GetProject(resourceProject types.String, config *transport_tpg.Config, diags *diag.Diagnostics) types.String {
```

Comment:
This could be refactored so that WithGetProject expects to have access to Config:

```diff
type WithGetProject struct {
+ *transport_tpg.Config
}
```

and then the GetProject method doesn't need config as an argument:

```diff
// GetProject determines whether to use a project value from the resource/data source config or the provider-default project
-func (w *WithGetProject) GetProject(resourceProject types.String, config *transport_tpg.Config, diags *diag.Diagnostics) types.String {
+func (w *WithGetProject) GetProject(resourceProject types.String, diags *diag.Diagnostics) types.String {
    
-    ...do stuff using config...
+    ...do stuff using w.Config...

}
```

though this might obscure what the method is doing to users.